tst/com/amazon/corretto/crypto/provider/test/AesCbcTest.java

```diff
@@ -0,0 +1,894 @@
+    final Cipher cbcPadding = accpAesCbcCipher(true);
+    cbcPadding.init(Cipher.ENCRYPT_MODE, key, iv);
+    final byte[] cipherText = cbcPadding.doFinal(input);
+    assertEquals(
+        "F29000B62A499FD0A9F39A6ADD2E77809543B86FC046FA883A9446B82E47D12DA144FC255AAD45BF681D3A3773A325C275C285C2760F0ED66EB65CFBEED8781D",
+        Hex.encodeHexString(cipherText, false));
+    cbcPadding.init(Cipher.DECRYPT_MODE, key, iv);
+    final ByteBuffer plainTextByteBuffer =
+        multiStepArray(
+            cbcPadding, Arrays.asList(0, 1, 1, 5, 5, 5, 5, 5, 5, 5, 5, 5, 5), cipherText);
+    final byte[] plainText =
+        Arrays.copyOf(plainTextByteBuffer.array(), plainTextByteBuffer.limit());
+    assertArrayEquals(input, plainText);
+  }
+
+  @Test
+  public void testOneShotPaddingDirect() throws Exception {
+    final SecretKeySpec key =
+        new SecretKeySpec(
+            TestUtil.decodeHex("000102030405060708090A0B0C0D0E0F101112131415161718191A1B1C1D1E1F"),
+            "AES");
+    final byte[] input =
+        TestUtil.decodeHex(
+            "000102030405060708090A0B0C0D0E0F101112131415161718191A1B1C1D1E1F202122232425262728292A2B2C2D2E2F3031");
+    final ByteBuffer directInput = ByteBuffer.allocateDirect(input.length);
+    directInput.put(input).flip();
+    final IvParameterSpec iv =
+        new IvParameterSpec(TestUtil.decodeHex("000102030405060708090A0B0C0D0E0F"));
+    final Cipher cbcPadding = accpAesCbcCipher(true);
+    cbcPadding.init(Cipher.ENCRYPT_MODE, key, iv);
+    final ByteBuffer cipherText = ByteBuffer.allocate(input.length + 16);
+    final int cipherLen = cbcPadding.doFinal(directInput, cipherText);
+    cipherText.flip();
+    assertEquals(cipherLen, cipherText.remaining());
+    assertEquals(
+        "F29000B62A499FD0A9F39A6ADD2E77809543B86FC046FA883A9446B82E47D12DA144FC255AAD45BF681D3A3773A325C275C285C2760F0ED66EB65CFBEED8781D",
+        Hex.encodeHexString(Arrays.copyOf(cipherText.array(), cipherText.remaining()), false));
+  }
+
+  @Test
+  public void testOneShotPaddingReadOnly() throws Exception {
+    final SecretKeySpec key =
+        new SecretKeySpec(
+            TestUtil.decodeHex("000102030405060708090A0B0C0D0E0F101112131415161718191A1B1C1D1E1F"),
+            "AES");
+    final byte[] input =
+        TestUtil.decodeHex(
+            "000102030405060708090A0B0C0D0E0F101112131415161718191A1B1C1D1E1F202122232425262728292A2B2C2D2E2F3031");
+    final ByteBuffer inputBuffer = ByteBuffer.allocate(input.length);
+    inputBuffer.put(input).flip();
+    final IvParameterSpec iv =
+        new IvParameterSpec(TestUtil.decodeHex("000102030405060708090A0B0C0D0E0F"));
+    final Cipher cbcPadding = accpAesCbcCipher(true);
+    cbcPadding.init(Cipher.ENCRYPT_MODE, key, iv);
+    // final ByteBuffer cipherText = ByteBuffer.allocate(input.length + 16);
+    // final int cipherLen = cbcPadding.doFinal(directInput, cipherText);
+    // cipherText.flip();
+    // assertEquals(cipherLen, cipherText.remaining());
```

Comment:
Should this code be uncommented/deleted?

Reply:
My bad. I forgot to remove the commented code.